arcgis-ios-sdk-samples/Maps/Manage operational layers/MMLLayersViewController.swift

```diff
@@ -15,77 +15,82 @@
 import UIKit
 import ArcGIS
 
-protocol MMLLayersViewControllerDelegate: AnyObject {
-    func layersViewControllerWantsToClose(_ layersViewController: MMLLayersViewController, withDeletedLayers layers: [AGSLayer])
-}
-
 class MMLLayersViewController: UITableViewController {
-    var layers: NSMutableArray!
-    var deletedLayers: [AGSLayer]!
     
-    weak var delegate: MMLLayersViewControllerDelegate?
+    /// Every layer on the map or that could be added to the map.
+    var allLayers: [AGSLayer] = []
+    
+    /// The layers attached to the map.
+    var operationalLayers: NSMutableArray?
+    
+    /// The layers present in `allLayers` but not in `operationalLayers`.
+    private var deletedLayers: [AGSLayer] {
+        if let operationalLayers = operationalLayers as? [AGSLayer] {
+            return allLayers.filter({ (layer) -> Bool in
+                return !operationalLayers.contains(layer)
+            })
+        }
+        return []
+    }
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        self.tableView.isEditing = true
+        tableView.isEditing = true
     }
     
-    func dataSourceIndexForIndexPath(_ dataSource: NSMutableArray, indexPath: IndexPath) -> Int {
+    private func dataSourceIndexForIndexPath(_ dataSource: NSMutableArray, indexPath: IndexPath) -> Int {
         return dataSource.count - indexPath.row - 1
     }
     
-    enum Section: Int {
+    private enum Section: Int {
         case added
         case removed
-        
-        init(_ rawValue: Int) {
```

Comment:
You prefer to force unwrap at each call site rather than in this initializer? This seemed like a pretty reasonable convenience to have. In all cases the table view is providing the section, providing the security that the passed in section will be a valid raw value.

Reply:
I guess I don't care either way. I changed it because I'd never seen this pattern before and it seemed nominally less safe.

Reply:
Taking a look at all the places where you had to add `rawValue: ` and `!`, I don't think removing this was an improvement.